### DISCUSSION TITLE
How to View iModal uploaded via services app in spa app

### DISCUSSION THREAD

**Discussion by afrazahmad-arthur** [iTwin/itwinjs-core#2332] [Q&A]
Hi , I have created two applications.

1- SPA app
2- services app

I am using a python server using services credentials to upload imodels to blob URL. I have a front end in react app and want to view iModal there.

In my react I am using spa credentials. but when I view imodal, I get  permission denied error 

![image](https://user-images.githubusercontent.com/77722198/134374351-1871c36b-9e49-4130-ab67-a5511a750461.png)

How can I display iModal uploaded via services app ?

**Comment by rseydler** [comment]
First I would check using https://developer.bentley.com/apis/imodels/operations/get-imodel-user-permissions/ to see what permissions are on that iModel GUID (4359....) for that user login. You can use the Try it Out button and login as the user when setting the authorization to authorizationCode.

If you don't get a permission set then you will need to add the user to the context (project) and provide permissions. https://developer.bentley.com/apis/projects/
